src/main/index.ts

```diff
@@ -271,30 +271,57 @@ app.on('web-contents-created', (_ev, contents) => {
     contents.session.storagePath &&
     contents.session.storagePath.indexOf('webxdc_') !== -1
   if (is_webxdc) {
-    contents.on('will-navigate', (e, navigationUrl) => {
+    const webxdc_open_url = (url: string) => {
+      if (url.startsWith('mailto:') || url.startsWith('openpgp4fpr:')) {
+        // handle mailto in dc
+        open_url(url)
+        mainWindow.window?.show()
+      }
+    }
+
+    contents.on('will-navigate', (ev, navigationUrl) => {
       if (navigationUrl.startsWith('webxdc://')) {
         // allow internal webxdc nav
         return
       } else if (navigationUrl.startsWith('mailto:')) {
         // handle mailto in dc
-        e.preventDefault()
-        open_url(navigationUrl)
-        mainWindow.window?.show()
+        ev.preventDefault()
+        webxdc_open_url(navigationUrl)
+      } else {
+        // prevent navigation to unknown scheme
+        ev.preventDefault()
+      }
+    })
+    
+    contents.on('will-frame-navigate', ev => {
+      if (ev.url.startsWith('webxdc://')) {
+        // allow internal webxdc nav
+        return
+      } else if (ev.url.startsWith('mailto:')) {
+        // handle mailto in dc
+        ev.preventDefault()
+        webxdc_open_url(ev.url)
       } else {
         // prevent navigation to unknown scheme
-        e.preventDefault()
+        ev.preventDefault()
       }
     })
+
+    contents.setWindowOpenHandler(_details => {
```

Comment:
Again, more a newbie question: 

I can see how this is the only path to handle `openpgp4fpr` links from here.

I understand why we need to prevent new windows of being created, but how come we're handling URLs in here and not only in the `will-navigate` or `will-frame-navigate` event handlers? Is there a case where links will only be handled here?

Reply:
Okay cool, thanks for clarifying!